starter-code/src/StackDataStructure.js

```diff
@@ -1,2 +1,34 @@
 function StackDataStructure () {
+  var that = this;
+  this.stackControl = [];
+  this.MAX_SIZE = 8;
+  this.isEmpty = function () {
+    if (that.stackControl.length === 0) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+  this.canPush = function () {
+    if (that.stackControl.length !== that.MAX_SIZE) {
```

Comment:
Better use this.stackControl.length < this.MAX_SIZE
Furthermore, you can return that without the if (up to you).